File: Android/android-exercise/JetPack/src/main/java/com/jet/pack/lifecycle/LifecycleActivity.java
package com.jet.pack.lifecycle;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.base.core.BaseAppCompatActivity;
import com.jet.pack.R;

public class LifecycleActivity extends BaseAppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.lifecycle_activity_main);

        initToolbarBringBack("Lifecycle");

        Java8Observer observer = new Java8Observer();
        getLifecycle().addObserver(observer);
    }


}

File: Kotlin/kotlin-android/app/src/main/java/com/nsz/kotlin/aac/architecture/lifecycle/LifecycleActivity.kt
package com.nsz.kotlin.aac.architecture.lifecycle

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class LifecycleActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        LifecycleHandler(this)
    }

}